crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -543,7 +549,15 @@ where
                         &alias.name.id
                     };
 
+                    // Look for imports `from __future__ import annotations`, ignore `as ...`
+                    self.has_future_annotations |= alias.name.id.as_str() == "annotations"
```

Comment:
Something like this:

```suggestion
                    // Look for imports `from __future__ import annotations`, ignore `as ...`.
                    // We intentionally don't enforce the rules about location of `__future__` imports here,
                    // we assume the user's intent was to apply the `__future__` import, so we still check
                    // using it (and will also emit a diagnostic about a mis-placed `__future__` import.)
                    self.has_future_annotations |= alias.name.id.as_str() == "annotations"
```